detox/src/android/espressoapi/DetoxAssertion.js

```diff
@@ -0,0 +1,133 @@
+/**
+
+	This code is generated.
+	For more information see generation/README.md.
+*/
+
+
+const log = require('../../utils/logger').child({ __filename });
+function sanitize_matcher(matcher) {
+  if (!matcher._call) {
+    return matcher;
+  }
+
+  const originalMatcher = typeof matcher._call === 'function' ? matcher._call() : matcher._call;
+  return originalMatcher.type ? originalMatcher.value : originalMatcher;
+} 
+class DetoxAssertion {
+  static assertMatcher(i, m) {
+    if (typeof m !== 'object' || typeof m.constructor !== 'function' || m.constructor.name.indexOf('Matcher') === -1) {
+      const isObject = typeof m === 'object';
+      const additionalErrorInfo = isObject ? typeof m.constructor === 'object' ? 'the constructor is no object' : 'it has a wrong class name: "' + m.constructor.name + '"' : 'it is no object';
+      log.error('m should be an instance of Matcher, got "' + m + '", it appears that ' + additionalErrorInfo);
+    }
+
+    return {
+      target: {
+        type: "Class",
+        value: "com.wix.detox.espresso.DetoxAssertion"
+      },
+      method: "assertMatcher",
+      args: [{
+        type: "ViewInteraction",
+        value: i
+      }, {
+        type: "Invocation",
+        value: sanitize_matcher(m)
+      }]
+    };
+  }
+
+  static assertNotVisible(i) {
+    return {
+      target: {
+        type: "Class",
+        value: "com.wix.detox.espresso.DetoxAssertion"
+      },
+      method: "assertNotVisible",
+      args: [{
+        type: "ViewInteraction",
+        value: i
+      }]
+    };
+  }
+
+  static assertNotExists(i) {
+    return {
+      target: {
+        type: "Class",
+        value: "com.wix.detox.espresso.DetoxAssertion"
+      },
+      method: "assertNotExists",
+      args: [{
+        type: "ViewInteraction",
+        value: i
+      }]
+    };
+  }
+
+  static waitForAssertMatcher(i, m, timeoutSeconds) {
+    if (typeof m !== 'object' || typeof m.constructor !== 'function' || m.constructor.name.indexOf('Matcher') === -1) {
+      const isObject = typeof m === 'object';
+      const additionalErrorInfo = isObject ? typeof m.constructor === 'object' ? 'the constructor is no object' : 'it has a wrong class name: "' + m.constructor.name + '"' : 'it is no object';
+      log.error('m should be an instance of Matcher, got "' + m + '", it appears that ' + additionalErrorInfo);
```

Comment:
Also, I see that you have almost or 100% identical error messages here, on lines 22, 73,  100. It's not easy to understand where the error messages come from. I suggest adding extra fields to the log message, e.g.:

![screenshot from 2018-08-13 10-37-55](https://user-images.githubusercontent.com/1962469/44018623-0c41afbc-9ee5-11e8-90a5-b3b402ccb1ad.png)

Below is a draft of what I think it is nice to have. It may be quick and dirty, and lacking generating `const util = require('util')` in the header

```diff
diff --git a/generation/core/generator.js b/generation/core/generator.js
index 86770566..ab97efe2 100644
--- a/generation/core/generator.js
+++ b/generation/core/generator.js
@@ -205,9 +205,9 @@ module.exports = function getGenerator({
     const typeCheckCreator = typeCheckInterfaces[type];
     const isListOfChecks = typeCheckCreator instanceof Array;
     return isListOfChecks
-      ? typeCheckCreator.map((singleCheck) => singleCheck(json))
+      ? typeCheckCreator.map((singleCheck) => singleCheck(json, functionName))
       : typeof typeCheckCreator === 'function'
-        ? typeCheckCreator(json)
+        ? typeCheckCreator(json, functionName)
         : t.emptyStatement();
   }
 
diff --git a/generation/core/type-checks.js b/generation/core/type-checks.js
index ccbe7e4a..764904f4 100644
--- a/generation/core/type-checks.js
+++ b/generation/core/type-checks.js
@@ -69,7 +69,7 @@ if (
 }
 
 function isOfClass(className) {
-  return ({ name }) =>
+  return ({ name }, functionName) =>
     template(`
 	if (
 		typeof ARG !== 'object' ||
@@ -79,10 +79,11 @@ function isOfClass(className) {
 		const isObject = typeof ARG === 'object';
 		const additionalErrorInfo = isObject ? (typeof ARG.constructor === 'object' ? 'the constructor is no object' : 'it has a wrong class name: "' + ARG.constructor.name +'"') : 'it is no object';
 
-		log.error('${name} should be an instance of ${className}, got "' + ARG + '", it appears that ' + additionalErrorInfo);
+		log.error({ event: EVENT, err: util.inspect(ARG) }, '${name} should be an instance of ${className}, got "' + ARG + '", it appears that ' + additionalErrorInfo);
 	}
 	`)({
-      ARG: t.identifier(name)
+      ARG: t.identifier(name),
+      EVENT: t.stringLiteral(functionName),
     });
 }
 
```

@DanielMSchmidt , I don't insist on it, however, but I'd be happy to know what you think about it.

Reply:
I'll give it a try. I like it better then before, but I would actually love to have an error there. But I think for that we would need to really check the integrity of the structure passed, to be sure we don't throw too eagerly. I will tackle this in a separate PR. For now, I'll change it to the better console.log